lib/puppet/provider/ios_aaa_authentication/cisco_ios.rb

```diff
@@ -71,7 +97,7 @@ def get(context)
   def set(context, changes)
     changes.each do |name, change|
       is = change.key?(:is) ? change[:is] : (get(context) || []).find { |key| key[:name] == name }
-      should = change[:should]
+      should = PuppetX::CiscoIOS::Utility.device_safe_instance(change[:should], commands_hash)
```

Comment:
If `canonicalize` is set up correctly as a feature then this wouldn't be needed and `should = change[:should]` will be enough as the `device_safe_instance` will be performed in the `canonicalize`

Reply:
As you have added the `canonicalize` feature then this can simply be `should = change[:should]`